### ISSUE TITLE
epub2txt: Can't open file '/tmp/epub2txt522957/META-INF/container.xml' for reading: No such file or directory

### ISSUE THREAD

**Issue by bekzatsk** [kevinboone/epub2txt2#29]
[books1.epub.zip](https://github.com/user-attachments/files/16989663/books1.epub.zip)

this file can't open

**Comment by kevinboone** [comment]
What format is the flle books1.epub.zip? From the name, it looks as if it ought to be a zip-compressed EPUB file, but it isn't. It looks like an EPUB that's been unpacked into a directory "books1.epub" and then repacked as a zipfile. If I unpack your file, then zip the contents of the 'books1.epub' directory _without_ the directory prefix, I end up with an EPUB that epub2txt can read fine. At least, it doesn't produce any errors -- I don't know if it's fine because I think it's in Russian, which I can't read :)

In short, I don't think that books1.epub.zip is a valid EPUB file, and it doesn't unzip into one, either. Is there any software that can read this as a EPUB? I'm sure there are smarter programs, that can figure out its structure, but epub2txt assumes a specification-compliant EPUB.

I'm sorry if I'm missing something.

**Comment by bekzatsk** [comment]
> What format is the flle books1.epub.zip? From the name, it looks as if it ought to be a zip-compressed EPUB file, but it isn't. It looks like an EPUB that's been unpacked into a directory "books1.epub" and then repacked as a zipfile. If I unpack your file, then zip the contents of the 'books1.epub' directory _without_ the directory prefix, I end up with an EPUB that epub2txt can read fine. At least, it doesn't produce any errors -- I don't know if it's fine because I think it's in Russian, which I can't read :)
> 
> 
> 
> In short, I don't think that books1.epub.zip is a valid EPUB file, and it doesn't unzip into one, either. Is there any software that can read this as a EPUB? I'm sure there are smarter programs, that can figure out its structure, but epub2txt assumes a specification-compliant EPUB.
> 
> 
> 
> I'm sorry if I'm missing something.
> 
> 

No, I ziped epub file because I can't upload to github issues.)

**Comment by kevinboone** [comment]
OK. But I did this:

unzip books1.epub.zip
cd books1.epub
zip -r /tmp/books1.epub .
.epub2txt /tmp/books1.epub

and it worked fine. The error message you get is compatible with the epub file having an extra directory at the top level, which is what I see.

Could you make the original EPUB file available in some other way, like Google Drive? Or just email it to me.

**Comment by kevinboone** [comment]
Sorry, without a way to reproduce the problem, I don't think I can fix it.
